sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LoggingEventBuilder.java

```diff
@@ -0,0 +1,349 @@
+
+    /*
+     * Performs the logging.
+     *
+     * @param format format-able message.
+     * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
+     */
+    void performLogging(LogLevel logLevel, String format, Object... args) {
+
+        Throwable throwable = null;
+        if (doesArgsHaveThrowable(args)) {
+            Object throwableObj = args[args.length - 1];
+
+            // This is true from before but is needed to appease SpotBugs.
+            if (throwableObj instanceof Throwable) {
+                throwable = (Throwable) throwableObj;
+            }
+
+            /*
+             * Environment is logging at a level higher than verbose, strip out the throwable as it would log its
+             * stack trace which is only expected when logging at a verbose level.
+             */
+            if (!logger.isDebugEnabled()) {
+                args = removeThrowable(args);
+            }
+        }
+
+        FormattingTuple tuple = MessageFormatter.arrayFormat(format, args);
+        String message = getMessageWithContext(tuple.getMessage(), throwable);
+
+        switch (logLevel) {
+            case VERBOSE:
+                logger.debug(message, tuple.getThrowable());
+                break;
+            case INFORMATIONAL:
+                logger.info(message, tuple.getThrowable());
+                break;
+            case WARNING:
+                logger.warn(message, tuple.getThrowable());
+                break;
+            case ERROR:
+                logger.error(message, tuple.getThrowable());
+                break;
+            default:
+                // Don't do anything, this state shouldn't be possible.
+                break;
+        }
+    }
+
+    /**
+     * Key value pair with basic serialization capabilities.
+     */
+    private static final class ContextKeyValuePair {
+        private final String key;
+        private final Object value;
+        private final Supplier<String> valueSupplier;
+
+        ContextKeyValuePair(String key, Object value) {
+            this.key = key;
+            this.value = value;
+            this.valueSupplier = null;
+        }
+
+        ContextKeyValuePair(String key, Supplier<String> valueSupplier) {
+            this.key = key;
+            this.value = null;
+            this.valueSupplier = valueSupplier;
+        }
+
+        /**
+         * Writes {"key":"value"} json string to provided StringBuilder.
+         */
+        public StringBuilder writeKeyAndValue(StringBuilder formatter) {
+            formatter.append("\"");
+            JSON_STRING_ENCODER.quoteAsString(key, formatter);
+            formatter.append("\":");
+
+            String valueStr = null;
+            if (value != null) {
+                // LoggingEventBuilder only supports primitives and strings
+                if (!(value instanceof String)) {
+                    JSON_STRING_ENCODER.quoteAsString(value.toString(), formatter);
```

Comment:
Will this result in `long`, `bool` types formatted as string? `"key": "false"` instead of `"key": false`

Reply:
nope, I have a test for it as well